plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -306,6 +321,45 @@ func (a *Elasticsearch) GetIndexName(indexName string, eventTime time.Time) stri
 		indexName = dateReplacer.Replace(indexName)
 	}
 
+	startTag := strings.Index(indexName, "{{")
+
+	for startTag >= 0 {
+
+		endTag := strings.Index(indexName, "}}")
+
+		if endTag < 0 {
+
+			startTag = -1
+
+		} else {
+
+			tagName := indexName[startTag+2 : endTag]
+			tagNameTrim := strings.TrimSpace(tagName)
+			found := false
+
+			for k := range metricTags {
```

Comment:
Since this happens for each metric, I'm concerned about the performance.  Perhaps this can be improved with by doing some of the work at Connect time.

I'll try to explain in more detail, this part needs only done once:
- Make an array of tagkeys: `["host"]`
- Build a format string from the indexName template with the tags replaced with %s: `telegraf-%s-%Y.%m.%d`

Then at Write time:
- Lookup the tag values and place them into an array.
- Use fmt.Sprintf with the format string and the array of tag values
- Use the existing replacer for the dates.